contrib/firmware-packager/firmware-packager.py

```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ * Copyright (C) 2017 Max Ehrlich max.ehr@gmail.com
+ *
+ * Licensed under the GNU General Public License Version 2
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import argparse
+import subprocess
+import contextlib
+import os
+import shutil
+import tempfile
+import time
+
+
+@contextlib.contextmanager
+def cd(path):
+    prev_cwd = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(prev_cwd)
+
+firmware_metainfo_template = """
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="firmware">
+  <id>{firmware_id}</id>
+  <name>{firmware_name}</name>
+  <summary>{firmware_summary}</summary>
+  <description>
+    {firmware_description}
+  </description>
+  <provides>
+    <firmware type="flashed">{device_unique_id}</firmware>
+  </provides>
+  <url type="homepage">{firmware_homepage}</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>proprietary</project_license>
+  <updatecontact>{contact_info}</updatecontact>
+  <developer_name>{developer_name}</developer_name>
+  <releases>
+    <release version="{release_version}" timestamp="{timestamp}">
+      <description>
+        {release_description}
+      </description>
+    </release>
+  </releases>
+</component>
+"""
+
+
+def make_firmware_metainfo(firmware_info, dst):
+    firmware_metainfo = firmware_metainfo_template.format(**vars(firmware_info), timestamp=time.time())
```

Comment:
I like the way you fill out this template.  I've not seen that in the past with python.